### PR TITLE
WIP: TST: pass random names in tm.makeDateIndex

### PR DESCRIPTION
**Pull request by jbrockmendel**
@WillAyd nearly all of the remaining test failures locally are in the json tests.  can you take a look and suggest whether we just need to pass `check_names=False`

I expect if we do something like this we won't want to actually randomize; that part I guess is a proof of concept for now.

### CODE REVIEW

**Comment by WillAyd** [comment]
Just at a brief glance not all of the JSON formats preserve index names. Split is a particular example which came up in #17393

**Comment by gfyoung** [comment]
Just curious: what was the motivation for adding this?

**Comment by jbrockmendel** [comment]
> Just curious: what was the motivation for adding this?

We are using these functions for fixtures that are aimed at trying to cover a wide swath of possible index objects, but are missing large parts of that space.

File: pandas/tests/series/test_apply.py [hunk 1]
```diff
@@ -520,7 +520,8 @@ def test_map_empty(self, index):
         result = s.map({})
 
         expected = pd.Series(np.nan, index=s.index)
-        tm.assert_series_equal(result, expected)
+        tm.assert_series_equal(result, expected, check_names=False)
+        # TODO: can we check names her?
```
Comment (gfyoung) [inline pandas/tests/series/test_apply.py]:
```suggestion
        # TODO: can we check names here?
```

**Comment by jbrockmendel** [comment]
closing to clear the queue, will keep trimming away at these locally

**Pull request closed without merging**